### PR TITLE
feat(ast) Remove the alias parameter from the Matching api

### PR DESCRIPTION
**Pull request by fpacifici**
We should never try to match an alias with anything (whether it is a string or not).
Aliases are an implementation detail of query processing that is meant to ensure the clickhouse query is not ambiguous. As such they may be replaced and changed in any way by the query processing pipeline while keeping the query intact.
They are not a structural part of the query we should rely on to identify constructs or patterns.

This removes the parameter entirely from the matcher api since using it would mean we are doing something probably broken. It also makes the API more concise.

### CODE REVIEW

File: snuba/datasets/discover.py [hunk 1]
```diff
@@ -72,15 +72,9 @@
 logger = logging.getLogger(__name__)
 
 EVENT_CONDITION = FunctionCallMatch(
-    None,
     Param("function", Or([StringMatch(op) for op in BINARY_OPERATORS])),
     (
-        Or(
-            [
-                ColumnMatch(None, None, StringMatch("type")),
-                LiteralMatch(StringMatch("type"), None),
```
Comment (fpacifici) [inline snuba/datasets/discover.py:81]: @evanh  do you know why we were trying to match an alias here ? It is the only palce in the code base.

**Comment by lynnagara** [comment]
Looks good to me - much simpler. Only question is - is there any chance at all that Sentry is sending any query that makes use of this alias that we will be breaking?

**Comment by fpacifici** [comment]
Are you talking specifically about that `type` alias in discover? Or in general? In general the api now ignores aliases, so an expression would match irrespective to whether the alias is there or not. Which is the behavior I would expect.

**Comment by lynnagara** [comment]
I was wondering about both, so good to know that we were always ignoring it, except for the potential exception of Discover `type`.

**Review by evanh — APPROVED** [review]
This is a solid change. See my comment about the discover use of aliases.

File: snuba/datasets/discover.py (hunk 1 above)
Comment (fpacifici) [reply snuba/datasets/discover.py]: Sentry tests are fine

**Pull request merged**